src/Aspire.Dashboard/Components/Pages/ConsoleLogs.razor.cs

```diff
@@ -600,6 +607,21 @@ private async Task ClearConsoleLogs(ApplicationKey? key)
         await ConsoleLogsManager.UpdateFiltersAsync(_consoleLogFilters);
     }
 
+    private Task IsPausedChangedAsync(bool isPaused)
+    {
+        PauseManager.ConsoleLogsPaused = isPaused;
+        if (isPaused)
+        {
+            PauseManager.ConsoleLogsPausedRanges.Add(new DateTimeRange(start: DateTime.UtcNow, end: null));
+        }
+        else
+        {
+            PauseManager.ConsoleLogsPausedRanges.Last().End = DateTime.UtcNow;
+        }
```

Comment:
Hmmm, I worry about thread safety of `ConsoleLogsPausedRanges`

How about making ConsoleLogsPaused have a private setter, and then add a `SetConsoleLogsPaused(bool isPaused)` method. Inside that method you would update the ranges as needed. Just encapsulates the code logic a little more.

Also, the method I mentioned [here](https://github.com/dotnet/aspire/pull/8184/files#r2006604953) could be on the PauseManager and the pause ranges collection could be made internal.

Instead of updating the list with a new value, I think a new collection should be created when a pause starts. That prevents any problem of the collection being iterated over while it is changed.